kolibri/plugins/coach/api.py

```diff
@@ -98,4 +98,10 @@ def get_queryset(self):
             Q(progress__gt=0) | Q(masterylogs__in=attempted_mastery_logs),
             user__in=get_members_or_user(self.kwargs['collection_kind'], self.kwargs['collection_id']),
             end_timestamp__gte=datetime_cutoff).values_list('content_id')
-        return ContentNode.objects.filter(content_id__in=recent_content_items)
+        # note from rtibbles:
+        # Possibly a horrible hack to ensure that we only return unique content_id'ed ContentNodes
+        # from the coach recent report endpoint.
+        # Would have loved to use distinct('content_id'), but unfortunately DISTINCT ON is Postgresql only
+        return ContentNode.objects.filter(pk__in=ContentNode.objects.filter(
+            content_id__in=recent_content_items).values('content_id').annotate(pk=Min('pk')).values_list(
+            'pk', flat=True))
```

Comment:
(Thanks for adding the note, @indirectlylit. I now understand the purpose.)

Wracked my brains, but can't think of an easier way either, without adding some pre-annotation to the DB during publishing to indicate which node is "primary" for each content_id. I think it could perhaps be improved in terms of readability, though, by splitting it into two lines:
```
pks_with_unique_content_ids = ContentNode.objects.filter(content_id__in=recent_content_items).values('content_id').annotate(pk=Min('pk')).values_list('pk', flat=True)
return ContentNode.objects.filter(pk__in=pks_with_unique_content_ids)
```

One nice thing is that it *does* combine them into a single query, so it could be worse in terms of efficiency.